services/group.go

```diff
@@ -113,43 +131,55 @@ func UpdateGroupById(id string, data *models.UpdateGroupRequest) (*models.Group,
 	return group, http.StatusOK, nil
 }
 
-func ListGroups(name string, limit int, offset int) ([]types.GroupDetailResponse, *int64, error) {
-	var groupDetailsList []types.GroupDetailResponse
+func ListGroups(name string, limit int, offset int) (*[]map[string]interface{}, *int64, error) {
+	var groupDetailsList []models.Group
+
 	var totalGroups int64
 
 	query := db.DB.Model(&models.Group{})
+
 	if name != "" {
 		query = query.Where("name LIKE ?", "%"+name+"%")
 	}
 
-	query = query.Select(`
-		groups.id AS id,
-		groups.name AS name,
-		groups.image AS image,
-		groups.created_at AS created_at,
-		groups.updated_at AS updated_at,
-		COUNT(DISTINCT group_events.id) AS events_count,
-		COUNT(DISTINCT user_groups.id) AS members_count
-	`).
-		Joins("LEFT JOIN group_events ON groups.id = group_events.group_id").
-		Joins("LEFT JOIN user_groups ON groups.id = user_groups.group_id").
-		Group("groups.id, groups.name, groups.image, groups.created_at, groups.updated_at")
-
-	err := query.Offset(offset).Limit(limit).Scan(&groupDetailsList).Error
-
+	err := query.
+		Preload("Tags").
+		Preload("Events", func(db *gorm.DB) *gorm.DB {
+			return db.Select("id")
+		}).
+		Preload("Members", "id").
+		Offset(offset).
+		Limit(limit).
+		Find(&groupDetailsList).Error
 	if err != nil {
 		return nil, nil, err
 	}
 
+	var response []map[string]interface{}
+	for _, group := range groupDetailsList {
+		groupMap := map[string]interface{}{
+			"id":            group.ID,
+			"name":          group.Name,
+			"image":         group.Image,
+			"created_at":    group.CreatedAt,
+			"updated_at":    group.UpdatedAt,
+			"tags":          group.Tags,
+			"members_count": len(group.Members),
+			"events_count":  len(group.Events),
+		}
+		response = append(response, groupMap)
+	}
```

Comment:
What happens when group hits 1 million records?  This should be done from the DB layer

Reply:
I figured since we're paginating, it shouldn't be a problem looping 10 times, even though it's not the best approach

Reply:
True. 10 items won't be much of a big problem